azurerm/internal/services/web/resource_arm_app_service.go

```diff
@@ -377,6 +377,8 @@ func resourceArmAppServiceUpdate(d *schema.ResourceData, meta interface{}) error
 	// If `source_control` is defined, we need to set site_config.0.scm_type to "None" or we cannot update it
 	_, hasSourceControl := d.GetOk("source_control.0.repo_url")
 
+	scmType := web.ScmType("None")
```

Comment:
is there a constant for this?

```suggestion
	scmType := web.ScmTypeNone
```